src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,205 @@
+
+        errors += SparkMaxUtils.check(SparkMaxUtils.UnitConversions.setDegreesFromGearRatio(
+                pivotAbsoluteEncoder, IntakeConstants.INPUT_ABS_ENCODER_GEAR_RATIO));
+
+        return errors == 0;
+    }
+
+    public void initIntakeTalons() {
+        TalonFXConfigurator cfgFront = intakeTalonFront.getConfigurator();
+        TalonFXConfigurator cfgBack = intakeTalonFront.getConfigurator();
+
+        TalonFXConfiguration toApply = new TalonFXConfiguration();
+        toApply.CurrentLimits.SupplyCurrentLimit = IntakeConstants.INTAKING_TALONS_CURRENT_LIMIT_AMPS;
+        toApply.CurrentLimits.SupplyCurrentLimitEnable = true;
+        toApply.MotorOutput.NeutralMode = NeutralModeValue.Coast;
+
+        cfgFront.apply(toApply);
+        cfgBack.apply(toApply);
+
+        intakeTalonBack.setControl(intakeTalonBackFollower);
+    }
+
+    /**
+     * @return the difference between the current time and the last controlled time
+     *         by the timer
+     *         if the lastControlledTime is empty (first record), then return 20
+     *         milliseconds
+     */
+    public double getTimeDifference() {
+        if (lastControlledTime.isEmpty()) {
+            return Constants.PERIOD_TIME_SECONDS;
+        }
+
+        double currentTimestamp = Timer.getFPGATimestamp();
+        double timestampDifference = currentTimestamp - lastControlledTime.get();
+        lastControlledTime = Optional.of(currentTimestamp);
+
+        return timestampDifference;
+    }
+
+    /** Returns the cosine of the arm angle in degrees off of the horizontal. */
+    public double getCosineArmAngle() {
+        return Math.cos(
+                Units.degreesToRadians(
+                        pivotAbsoluteEncoder.getPosition() - IntakeConstants.INTAKE_POSITION_WHEN_HORIZONTAL_DEGREES));
+    }
+
+    public double getOffsetAbsPositionDeg() {
+        return pivotAbsoluteEncoder.getPosition() + IntakeCal.INTAKE_ABSOLUTE_ENCODER_ZERO_OFFSET_DEG;
+    }
+
+    public void moveToPos(IntakePosition pos) {
+        pivotController.setGoal(intakePositionMap.get(pos));
+        double intakeDemandVoltsA = pivotController.calculate(getOffsetAbsPositionDeg());
+
+        double intakeDemandVoltsC = IntakeCal.ARBITRARY_INTAKE_PIVOT_FEEDFORWARD_VOLTS * getCosineArmAngle();
+
+        double intakeDemandVoltsB;
+        double currentVelocity = pivotController.getSetpoint().velocity;
+        if (prevVelocityMetersPerSec.isEmpty()) {
+            intakeDemandVoltsB = IntakeCal.INTAKE_PIVOT_FEEDFORWARD.calculate(currentVelocity);
+        } else {
+            intakeDemandVoltsB = IntakeCal.INTAKE_PIVOT_FEEDFORWARD.calculate(prevVelocityMetersPerSec.get(),
+                    currentVelocity, getTimeDifference());
+        }
+
+        prevVelocityMetersPerSec = Optional.of(currentVelocity);
+
+        pivotMotor.setVoltage(intakeDemandVoltsA + intakeDemandVoltsB + intakeDemandVoltsC);
+    }
+
+    public boolean atDesiredIntakePosition() {
```

Comment:
This has repeated code from the below function. Can you do this without repeating code

Reply:
yupsies